src/App.js

```diff
@@ -101,10 +101,95 @@ class Board extends React.Component {
     return true;
   }
 
+  //Remove a random number from the board
+  removeRandomNumber(valueBoard) {
+
+    //Pick a random position o the board
+    let randomPositionOfBoard = Math.floor(Math.random() * valueBoard.length);
+
+    let backup = [];
+
+    //Change the random number if the position it's on was a zero
+    while(parseInt(valueBoard[randomPositionOfBoard]) === 0) {
+      //Reset random number
+      randomPositionOfBoard = Math.floor(Math.random() * valueBoard.length);
+    }
+
+    //Back up position of board's value removed and it's number
+    backup.push(randomPositionOfBoard);
+    backup.push(valueBoard[randomPositionOfBoard]);
+
+    //Set value of the random picked position to zero
+    valueBoard[randomPositionOfBoard] = 0;
+    
+    //Tell which position got their number removed and what number
+    return backup;
+  }
+
+  //Undo removeRandomNumber
+  undoRemoveRandomNumber(undo, valueBoard) {
+    //Reset the position, given from removeRandomNumber method, of the board's value
+    valueBoard[undo[0]] = undo[1];
+  }
+
+  findAllPossibleAnswers(valueBoard, backup) {
```

Comment:
If I understand correctly, I think this is getting all the possible numbers for each backup square. But when I read the name I thought this function would try to find all possible solutions to the Sudoku puzzle. I wouldn't use the word `answers` here in the function name

Reply:
Sorry for the bad names.....

Reply:
haha no worries!